core-bundle/src/DataContainer/DataContainerOperation.php

```diff
@@ -65,7 +65,7 @@ public function offsetExists(mixed $offset): bool
 
     public function offsetGet(mixed $offset): mixed
     {
-        return $this->operation[$offset];
+        return $this->operation[$offset] ?? null;
```

Comment:
What about line 78? And why is this necessary at all? Did you not call `offsetExists()` in advance?

Reply:
as far as I know, `unset()` would not throw a warning, same as `empty()` etc. (because they are not functions but language features). Contao I did not call `offsetExists` (which is actually `isset`), because we never do.

Reply:
`unset()`, `empty()`, `isset()` and `??` seem to behave exactly the same as an array would, they all check `offsetExists()` first, see https://3v4l.org/YC4it

I think you need to use `??` in the calling code to fix this.